modules/http/src/main/java/org/mule/module/http/internal/request/grizzly/GrizzlyHttpClient.java

```diff
@@ -627,7 +622,7 @@ public InputStream sendAndReceiveInputStream(HttpRequest request, int responseTi
     {
         Request grizzlyRequest = createGrizzlyRequest(request, responseTimeout, followRedirects, authentication);
         PipedOutputStream outPipe = new PipedOutputStream();
-        PipedInputStream inPipe = new PipedInputStream(outPipe);
+        PipedInputStream inPipe = new PipedInputStream(outPipe, responseBufferSize);
```

Comment:
Because it's 3 different send methods, blocking, non blocking and some stuff for ws consumer.